.github/workflows/cd.yaml

```
@@ -83,32 +83,31 @@ jobs:
 
   build:
     runs-on: ubuntu-latest
-    needs: [tests, linters]
-    if: github.ref == 'refs/heads/main'
+    permissions:
+      id-token: write # This is required for requesting the JWT
+      contents: read  # This is required for actions/checkout
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v3
-      - name: Store current date
-        run: echo "BUILD_DATE=$(date +%Y%m%d%H%M)" >> $GITHUB_ENV
-      - name: Build
-        run: |
-          docker build \
-            --label build.git.sha=${{ github.sha }} \
-            --label build.git.branch=${{ github.ref }} \
-            --label build.date=${{ env.BUILD_DATE }} \
-            --build-arg BUILD_DATE=${{ env.BUILD_DATE }} \
-            --build-arg BUILD_TAG=${{ github.ref }} \
-            --build-arg COMMIT_ID=${{ github.sha }} \
-            -t app .
-      - name: Push to ECR
-        id: ecr
-        uses: jwalton/gh-ecr-push@v1
+      # Checkout GitHub repository
+      - uses: actions/checkout@v3
+
+      # Assume role in Cloud Platform
+      - uses: aws-actions/configure-aws-credentials@v2
         with:
-          access-key-id: ${{ secrets.ECR_AWS_ACCESS_KEY_ID }}
-          secret-access-key: ${{ secrets.ECR_AWS_SECRET_ACCESS_KEY }}
-          region: eu-west-2
-          local-image: app
-          image: ${{ secrets.ECR_NAME }}:${{ github.sha }}
+          role-to-assume: ${{ secrets.ECR_ROLE_TO_ASSUME }}
+          aws-region: ${{ vars.ECR_REGION }}
+
+      # Login to container repository
+      - uses: aws-actions/amazon-ecr-login@v1
+        id: login-ecr
+
+      # Build and push a Docker image to the container repository
+      - run: |
+          docker build -t $REGISTRY/$REPOSITORY:gha-$IMAGE_TAG .
+          docker push $REGISTRY/$REPOSITORY:gha-$IMAGE_TAG
+        env:
+          REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          REPOSITORY: ${{ vars.ECR_REPOSITORY }}
```

You could maybe do something like:
```suggestion
          ECR_URL: ${{ steps.login-ecr.outputs.registry }}/${{ vars.ECR_REPOSITORY }}
```

and then use that above:
```
docker push $ECR_URL:$IMAGE_TAG
```

That way you don't need to change any of the deploy scripts.

---

Although I don't know if those env vars would then be available in the deploy section